app/src/main/java/com/example/plantsapp/data/RoomTasksRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.plantsapp.data
+
+import androidx.core.net.toUri
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.repository.TasksRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import java.util.Date
+
+class RoomTasksRepository(
+    private val plantsDao: RoomPlantsDao
+) : TasksRepository {
+
+    override suspend fun getTasksForDate(date: Date): Flow<List<Task>> {
+        return plantsDao.getAll()
+            .map { plants ->
+                plants.filter {
+                    ((date.time - it.lastWatered) / DAY_IN_MILLISECONDS) %
+                            it.wateringFrequencyDays == 0L
+                }
+                    .map {
+                        Task(
+                            // TODO remove task action setting from repository
+                            WATERING_ACTION,
+                            it.name,
+                            it.plantPicture?.toUri()
+                        )
+                    }
+            }
+    }
+}
```

Comment:
Lets simplify this code little bit: 
1. lastWatered -> creation date
2. move filtering condition inside of method with understandable name
3. time will be easier to read if 1000 * 60 *.....* 24